### PR TITLE
Add configuration migration self tests

### PR DESCRIPTION
**Pull request by superm1**
I don't think we've seen the migration end of the problems yet.
I started to write the tests and then realized that there were some cases that weren't covered (like an empty key that migrates with a comment above).

This PR makes some changes to migration and also adds the self tests.  A subset of it should come to 1_9_X, but let's get it right in main first and then figure out how much to bring back.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-config.c [hunk 1]
```diff
@@ -108,6 +108,54 @@ fu_config_emit_loaded(FuConfig *self)
...
+		const gchar *value;
+	} key_values[] = {
+	    {"fwupd", "ApprovedFirmware", ""},
+	    {"fwupd", "ArchiveSizeMax", "0"},
+	    {"fwupd", "BlockedFirmware", ""},
+	    {"fwupd", "DisabledDevices", ""},
+	    {"fwupd", "EnumerateAllDevices", "false"},
+	    {"fwupd", "EspLocation", ""},
+	    {"fwupd", "HostBkc", ""},
+	    {"fwupd", "IdleTimeout", FU_DAEMON_CONFIG_DEFAULT_IDLE_TIMEOUT_STR},
```
Comment (hughsie) [inline libfwupdplugin/fu-config.c:126]: I think you can use G_STRINGIFY here.

File: meson.build [hunk 2]
```diff
@@ -536,8 +536,19 @@ motd_dir = 'motd.d'
...
+conf.set_quoted('FU_DAEMON_CONFIG_DEFAULT_URI_SCHEMES', 'file;https;http;ipfs')
+conf.set_quoted('FU_DAEMON_CONFIG_DEFAULT_TRUSTED_REPORTS', 'VendorId=$OEM')
+conf.set_quoted('FU_DAEMON_CONFIG_DEFAULT_RELEASE_PRIORITY', 'local')
+conf.set_quoted('FU_MSR_CONFIG_DEFAULT_MINIMUM_SME_KERNEL_VERSION', '5.18.0')
+conf.set_quoted('FU_THUNDERBOLT_CONFIG_DEFAULT_MINIMUM_KERNEL_VERSION', '4.13.0')
+conf.set_quoted('FU_REDFISH_CONFIG_DEFAULT_MANAGER_RESET_TIMEOUT', '1800')
+conf.set('FU_DAEMON_CONFIG_DEFAULT_IDLE_TIMEOUT', 7200)
+conf.set_quoted('FU_DAEMON_CONFIG_DEFAULT_IDLE_TIMEOUT_STR', '7200')
+conf.set('FU_REDFISH_CONFIG_DEFAULT_IPMI_DISABLE_CREATE_USER', false)
+conf.set_quoted('FU_REDFISH_CONFIG_DEFAULT_IPMI_DISABLE_CREATE_USER_STR', 'False')
```
[the remark this answers is not in the document]
Comment (superm1) [reply meson.build]: changed

[the remark this answers is not in the document]
Comment (superm1) [reply meson.build]: fixed

**Comment by superm1** [comment]
closed in favor of #6766

**Pull request closed without merging**